pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -467,28 +473,55 @@ func (r *ReferenceAdapter) deleteCredentials() error {
 	return nil
 }
 
+type EnsureAzResult int
+
+const (
+	EnsureAzResultNotReady EnsureAzResult = iota
+	EnsureAzResultModified
+	EnsureAzResultNoChange
+)
+
 // ensureAvailibilityZonesSet sets the az in the projectclaim spec if necessary
 // returns true if the project claim has been modified
-func (r *ReferenceAdapter) ensureClaimAvailibilityZonesSet() (bool, error) {
+func (r *ReferenceAdapter) ensureClaimAvailibilityZonesSet() (EnsureAzResult, error) {
 	if len(r.ProjectClaim.Spec.AvailibilityZones) > 0 {
-		return false, nil
+		return EnsureAzResultNoChange, nil
 	}
 
 	zones, err := r.gcpClient.ListAvilibilityZones(r.ProjectReference.Spec.GCPProjectID, r.ProjectClaim.Spec.Region)
 	if err != nil {
-		return false, err
+		return r.handleAvailabilityZonesError(err)
 	}
+	conditions := &r.ProjectReference.Status.Conditions
+	r.conditionManager.SetCondition(conditions, gcpv1alpha1.ConditionComputeApiReady, corev1.ConditionTrue, "QueryAvailabilityZonesSucceeded", "ComputeAPI ready, successfully queried availability zones")
 
 	r.ProjectClaim.Spec.AvailibilityZones = zones
 
-	return true, nil
+	return EnsureAzResultModified, nil
+}
+
+func (r *ReferenceAdapter) handleAvailabilityZonesError(err error) (EnsureAzResult, error) {
+	conditions := &r.ProjectReference.Status.Conditions
+	if strings.HasPrefix(err.Error(), "googleapi: Error 403: Access Not Configured. Compute Engine API has not been used in project") {
```

Comment:
I hope google doesn't change this error message.

Reply:
unfortunately we don't have a type or anything to recognize the error, other than this message...

Reply:
maybe add a comment that explains it so if it changes we know it's fragile

Reply:
```
// This error message is not backed in any real error type, so working with string comparison is our best approach
```

Reply:
not sure a comment like this describes it better than the actual code

Reply:
tried to make that clear with a function